File: count_words.py
import os
import re

def count_words_in_file(filepath):
    """Count word entries in a TypeScript file"""
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Count occurrences of "word: '"
    count = len(re.findall(r"word:\s*'", content))
    return count

def main():
    words_dir = "packages/content/src/words"
    total = 0
    
    print("=== 词汇文件统计 ===")
    
    for filename in os.listdir(words_dir):
        if filename.endswith('.ts'):
            filepath = os.path.join(words_dir, filename)
            count = count_words_in_file(filepath)
            print(f"{filename}: {count}")
            total += count
    
    print(f"=== 总计 ===")
    print(f"总词汇数: {total}")
    
    # Check if we need more words
    target = 1000
    if total < target:
        print(f"还需要 {target - total} 个词")
    else:
        print(f"已达到目标 {target} 个词")

if __name__ == "__main__":
    main()

File: count_words.js
const fs = require('fs');
const path = require('path');

function countWordsInFile(filepath) {
    const content = fs.readFileSync(filepath, 'utf-8');
    // Count occurrences of "word: '"
    const matches = content.match(/word:\s*'/g);
    return matches ? matches.length : 0;
}

function main() {
    const wordsDir = path.join(__dirname, 'packages', 'content', 'src', 'words');
    let total = 0;
    
    console.log("=== 词汇文件统计 ===");
    
    const files = fs.readdirSync(wordsDir).filter(f => f.endsWith('.ts'));
    
    files.forEach(filename => {
        const filepath = path.join(wordsDir, filename);
        const count = countWordsInFile(filepath);
        console.log(`${filename}: ${count}`);
        total += count;
    });
    
    console.log("=== 总计 ===");
    console.log(`总词汇数: ${total}`);
    
    // Check if we need more words
    const target = 1000;
    if (total < target) {
        console.log(`还需要 ${target - total} 个词`);
    } else {
        console.log(`已达到目标 ${target} 个词`);
    }
}

main();